lib/github-pages-health-check/domain.rb

```diff
@@ -170,7 +174,11 @@ def dns
         return unless valid_domain?
         @dns = Timeout.timeout(TIMEOUT) do
           GitHubPages::HealthCheck.without_warnings do
-            Net::DNS::Resolver.start(absolute_domain).answer unless host.nil?
+            unless host.nil?
+              resolver = Net::DNS::Resolver.new
+              resolver.search(absolute_domain, Net::DNS::A).answer +
+                resolver.search(absolute_domain, Net::DNS::MX).answer
```

Comment:
This is now always going to make two DNS calls, even if we only need one. Would it make more sense to abstract this out into an `A` record check and an `MX` record check?